python/flink_agents/api/event.py

```diff
@@ -0,0 +1,59 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+#################################################################################
+
+from abc import ABC
+from typing import Any
+from uuid import UUID, uuid4
+
+from pydantic import BaseModel, Field
+
+
+class Event(BaseModel, ABC, extra="allow"):
+    """Base class for all event types in the system.
+
+    Attributes:
+    ----------
+    id : UUID
+        Unique identifier for the event, automatically generated using uuid4.
+    """
+
+    id: UUID = Field(default_factory=uuid4)
+
+
+class InputEvent(Event):
+    """Event generated by the framework, carrying an input data that
+    arrives at the workflow.
+    """
+
+    input: Any
+
+
+class OutputEvent(Event):
+    """Event representing a result from workflow. By generating an OutputEvent,
+    actions can emit output data.
+
+    Attributes:
+    ----------
+    isLegal : bool
+        Is the result legal or not.
+    result : str
+        The final result returned by the workflow.
+    """
+
+    isLegal: bool
```

Comment:
We should not need this. If an output is not legal, it should not be outputted at all.